config/routes.rb

```diff
@@ -7,10 +7,12 @@
 
   devise_for :users, controllers: { registrations: :registrations }
 
-  resources :keywords, only: [:index, :create]
+  resources :keywords, only: [:index, :create, :show]
 
   resources :filters, only: :index
 
+  resources :preview, only: :show
```

Comment:
For this kind of related route, In Rails we usually create nested routes like
```Ruby
  resources :keywords, only: [:index, :create, :show] do
    resource :preview, only: :show
  end
```
It will create a route `/keywords/:keyword_id/preview` .
(This comment is for improvement and optional)

Reply:
This is A-MA-ZING! :) I felt a little bit constraint by the convention to remain inside CRUD (create/index/show/...). But nesting routes really solve this!!! Thanks a lot!! 🤩

Reply:
Note that this changed:
1. prevew#show is not the 'index' action (because we do not want to specify the keyword_id a second time)
2. `ïd` param becomes `keyword_id` params (wow, Rails magic!!) :D